dev/release/verify-release-candidate.bat

```diff
@@ -112,6 +112,8 @@ cmake -G "%GENERATOR%" ^
       -DARROW_WITH_ZSTD=ON ^
       -DCMAKE_BUILD_TYPE=%CONFIGURATION% ^
       -DCMAKE_INSTALL_PREFIX=%ARROW_HOME% ^
+      -DCMAKE_PREFIX_PATH=%CONDA_PREFIX%\Lib ^
```

Comment:
I think that we don't need to append `\Lib` for `CMAKE_PREFIX_PATH`:

```suggestion
      -DCMAKE_PREFIX_PATH=%CONDA_PREFIX% ^
```

https://cmake.org/cmake/help/latest/variable/CMAKE_PREFIX_PATH.html

> Each command will add appropriate subdirectories (like `bin`, `lib`, or `include`) as specified in its own documentation.